### PR TITLE
Remove validation proxy

### PR DESCRIPTION
**Pull request by marcotc**
This PR removes the `ValidationProxy` concept, as we are moving towards a central `Datadog.configuration` settings instance, with namespaced options per-product.

This PR will be followed up by changes to the current options, namespacing `tracing`, `profiling`, `appsec` and `ci` settings.

### Breaking change

* Removed: `Settings` object argument for `Datadog.configure`.

### CODE REVIEW

**Comment by delner** [comment]
It might be too early to ask, but are we going to remove `Datadog#internal_configure` in favor of `#configure` in this PR?

I expect we're also going to remove `Datadog::Tracing#configure` but as you mentioned, it sounds like you're waiting to do that in another PR.

**Comment by marcotc** [comment]
I remember why I didn't want to do it now: I would have to update tests for Datadog::Profiling.configure/configuration and Datadog::Tracing.configure/configuration that won't exist two PRs down the line, so I was avoiding doing necessary work.

But I only realized this halfway through adding the extra changes, so intermediate tests are fixed.

File: lib/datadog/core/configuration.rb [hunk 1]
```diff
@@ -83,17 +80,22 @@ def configuration
...
       # any are applicable.
       #
-      # @param [Datadog::Core::Configuration::Settings] configuration the base configuration object. Provide a custom
-      #   instance if you are managing the configuration yourself. By default, the global configuration object is used.
       # @yieldparam [Datadog::Core::Configuration::Settings] c the mutable configuration object
-      def configure(configuration = self.configuration)
-        # Wrap block with global option validation
-        wrapped_block = proc do |c|
-          yield(ValidationProxy::Global.new(c))
+      def configure
```
Comment (marcotc) [inline lib/datadog/core/configuration.rb:84]: Not sure if this got reintroduced in 1.0, but the `configuration = self.configuration` has no valid public use (no valid internal use today either).

We used to allow an argument to be passed to `#configure`, but that use case has moved to `#configure_onto`.

Comment (delner) [reply lib/datadog/core/configuration.rb:84]: In theory should we permit a `Settings` object to be given to it? If not, or there's no real case for this, we can remove.

**Review by delner — APPROVED** [review]

**Pull request merged**